src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java

```diff
@@ -190,25 +206,52 @@ void saveEventPopulated(EventType eventType) {
         Assertions.assertEquals(dateStarted, savedOrDeletedBook.getDateStartedReading());
         Assertions.assertEquals(dateFinished, savedOrDeletedBook.getDateFinishedReading());
         Assertions.assertEquals(ratingVal, savedOrDeletedBook.getRating());
         Assertions.assertEquals(seriesPosition, savedOrDeletedBook.getSeriesPosition());
     }
 
-    private void populateBookForm() {
-        bookForm.authorFirstName.setValue(firstName);
-        bookForm.authorLastName.setValue(lastName);
-        bookForm.bookTitle.setValue(bookTitle);
-        bookForm.predefinedShelfField.setValue(readShelf.getPredefinedShelfName());
-        bookForm.bookGenre.setValue(genre);
-        bookForm.pagesRead.setValue(pagesRead);
-        bookForm.numberOfPages.setValue(numberOfPages);
-        bookForm.dateStartedReading.setValue(dateStarted);
-        bookForm.dateFinishedReading.setValue(dateFinished);
-        bookForm.rating.setValue(converter.convertToPresentation(ratingVal, null));
+    private void populateBookForm(ShelfName shelfName, boolean isInSeries) {
+        populateBookFormWithExistingBook(shelfName, isInSeries, null);
+    }
+
+    private void populateBookFormWithExistingBook(ShelfName shelfName, boolean isInSeries, Book maybePresentBook) {
+        if (maybePresentBook != null) {
+            bookForm.setBook(maybePresentBook);
+        } else {
+            bookForm.authorFirstName.setValue(firstName);
+            bookForm.authorLastName.setValue(lastName);
+            bookForm.bookTitle.setValue(bookTitle);
+            bookForm.predefinedShelfField.setValue(readShelf.getPredefinedShelfName());
+            bookForm.bookGenre.setValue(genre);
+            bookForm.numberOfPages.setValue(numberOfPages);
+            if (isInSeries) {
+                bookForm.seriesPosition.setValue(SERIES_POSITION);
+            }
+        }
+        switch (shelfName) {
+            case TO_READ:
+                bookForm.predefinedShelfField.setValue(shelfName);
+                break;
+            case READING:
+                bookForm.predefinedShelfField.setValue(shelfName);
+                bookForm.dateStartedReading.setValue(dateStarted);
+                break;
+            case READ:
+                bookForm.predefinedShelfField.setValue(shelfName);
+                bookForm.dateStartedReading.setValue(dateStarted);
+                bookForm.dateFinishedReading.setValue(dateFinished);
+                bookForm.rating.setValue(converter.convertToPresentation(ratingVal, null));
+                break;
+            case DID_NOT_FINISH:
+                bookForm.predefinedShelfField.setValue(shelfName);
+                bookForm.dateStartedReading.setValue(dateStarted);
+                bookForm.pagesRead.setValue(pagesRead);
+                bookForm.rating.setValue(converter.convertToPresentation(ratingVal, null));
+                break;
+        }
     }
```

Comment:
How about below refactoring?
```    
    private void populateBookForm(ShelfName shelfName, boolean isInSeries) {
        bookForm.authorFirstName.setValue(firstName);
        bookForm.authorLastName.setValue(lastName);
        bookForm.bookTitle.setValue(bookTitle);
        bookForm.predefinedShelfField.setValue(readShelf.getPredefinedShelfName());
        bookForm.bookGenre.setValue(genre);
        bookForm.numberOfPages.setValue(numberOfPages);
        if (isInSeries) {
            bookForm.seriesPosition.setValue(SERIES_POSITION);
        }
        populateBookShelf(shelfName);
    }

    private void populateBookFormWithExistingBook(ShelfName shelfName, Book maybePresentBook) {
        bookForm.setBook(maybePresentBook);
        populateBookShelf(shelfName);
    }

    private void populateBookShelf(ShelfName shelfName) {
        switch (shelfName) {
            case TO_READ:
                bookForm.predefinedShelfField.setValue(shelfName);
                break;
            case READING:
                bookForm.predefinedShelfField.setValue(shelfName);
                bookForm.dateStartedReading.setValue(dateStarted);
                break;
            case READ:
                bookForm.predefinedShelfField.setValue(shelfName);
                bookForm.dateStartedReading.setValue(dateStarted);
                bookForm.dateFinishedReading.setValue(dateFinished);
                bookForm.rating.setValue(converter.convertToPresentation(ratingVal, null));
                break;
            case DID_NOT_FINISH:
                bookForm.predefinedShelfField.setValue(shelfName);
                bookForm.dateStartedReading.setValue(dateStarted);
                bookForm.pagesRead.setValue(pagesRead);
                bookForm.rating.setValue(converter.convertToPresentation(ratingVal, null));
                break;
        }
    }
```

Reply:
Yes, I like that!
Thank you, it didn't occur to me to make a cut there, it makes the code way cleaner.